src/main/java/core/basesyntax/services/impl/ReportGeneratorServiceImpl.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax.services.impl;
+
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.services.ReportGeneratorService;
+
+public class ReportGeneratorServiceImpl implements ReportGeneratorService {
+    private final FruitDao fruitDao;
+
+    public ReportGeneratorServiceImpl(FruitDao fruitDao) {
+        this.fruitDao = fruitDao;
+    }
+
+    @Override
+    public String getReport() {
+        StringBuilder report = new StringBuilder();
+        report.append("fruit,quantity").append(System.lineSeparator());
+        fruitDao.getStorage().forEach((fruit, quantity) -> report.append(fruit)
+                .append(",")
+                .append(quantity)
+                .append(System.lineSeparator()));
+
+        return report.toString();
```

Comment:
The method 'getReport' is directly accessing the storage through 'fruitDao.getStorage()'. This could lead to potential issues if the storage structure changes. Consider adding a method in the FruitDao that abstracts the details of how the report data is retrieved.